### ISSUE TITLE
Reconnections fail after 3 attempts.

### ISSUE THREAD

**Issue by SeppoTakalo** [ARMmbed/mbed-client-example-6lowpan#37]
When running the application on 6LoWPAN network it re-creates the interface classes after each network failures, this causes the re-connections to start failing after 3 re-connections.

After this, I will constantly see:
~~~
[DBG ][mbedclient]: waiting 5s before sending registration...
[DBG ][mbedclient]: send_registration()
[ERR ][mbedclient]: error() 6
~~~

How to re-produce:
1. Modify application so that when receiving object_registered callback, you destroy the connection and restart the registration proses.
2. Run application.
3. Observe that after 3 succesful registrations, it starts to fail.

**Comment by SeppoTakalo** [comment]
CC: @yogpan01 @artokin @juhhei01

**Comment by yogpan01** [comment]
mbed client is using mbed OS socket and when any application deletes mbed client instance, it will call mbedSocket destructor, it is the responsibility of underlying porting layer to ensure that proper cleanup happens.

**Comment by SeppoTakalo** [comment]
Yes, of course. I just CC:d that to you also because you might be interested, and you should be familiar with this issue.
We continue the investigation to find to root cause for this.

**Comment by rainierwolfcastle** [comment]
ARM Internal Ref: ONME-2072

**Comment by artokin** [comment]
@AnttiKauppila , would you please help on this issue? It looks like there is something wrong in SSL handshake, in M2MConnectionHandlerPimpl::dns_handler, start_connecting_non_blocking() is returning error.

**Comment by AnttiKauppila** [comment]
Fixed in mbed-client-mbedtls version 1.0.8
